app/models/job.rb

```diff
@@ -1,4 +1,25 @@
 class Job < ActiveRecord::Base
-  validates :title, :level_of_interest, :city, presence: true
+  validates :title, :level_of_interest, :city, :category, presence: true
   belongs_to :company
-end
+  belongs_to :category
+  has_many :comments, :dependent => :nullify
+  
+
+  def self.sort_by(attribute)
+    return order(:city).reverse_order if attribute == "location"
+    return order(:level_of_interest).reverse_order if attribute == "interest"
+  end
+
+  def self.count_by_interest
+    group(:level_of_interest).count.sort_by {|k , v| v}.reverse[0..12]
```

Comment:
i encourage you to choose better names for `k` and `v`. these are not very descriptive